go/oasis-node/cmd/debug/txsource/workload/commission.go

```diff
@@ -0,0 +1,370 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasislabs/oasis-core/go/common/crypto/signature/signers/memory"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	consensus "github.com/oasislabs/oasis-core/go/consensus/api"
+	"github.com/oasislabs/oasis-core/go/consensus/api/transaction"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+	runtimeClient "github.com/oasislabs/oasis-core/go/runtime/client/api"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+)
+
+const (
+	// NameCommission is the name of the commission schedule amendements
+	// workload.
+	NameCommission = "commission"
+
+	commissionMaxBoundChangeEpochInterval = 10
+	commissionMaxRateChangeEpochInterval  = 10
+)
+
+type commission struct {
+	logger *logging.Logger
+
+	rules          staking.CommissionScheduleRules
+	account        signature.Signer
+	reckonedNonce  uint64
+	fundingAccount signature.Signer
+}
+
+// currentBound returns the rate bounds at the latest bound step that has
+// started or nil if no step has started.
+func currentBound(cs *staking.CommissionSchedule, now epochtime.EpochTime) *staking.CommissionRateBoundStep {
+	var latestStartedStep *staking.CommissionRateBoundStep
+	for i := range cs.Bounds {
+		step := &cs.Bounds[i]
+		if step.Start > now {
+			break
+		}
+		latestStartedStep = step
+	}
+	if latestStartedStep == nil {
+		return nil
+	}
+	return latestStartedStep
+}
+
+// genValidRateStep generates a commission rate step that conforts to all bound
+// rules between start and end epoch. The function panics in case a rate step
+// cannot satisfy all bound rules so the caller should make sure that bounds
+// are not exclusive.
```

Comment:
this panics in `Int63n` in this case. would we be better people if we did our own panic explicitly?

Reply:
probably yes, will add